Admin/CrossTenantMailboxMigrationValidation.ps1

```diff
@@ -0,0 +1,523 @@
+            Write-Host ">> Error: Organization relationship on SOURCE tenant MailboxMove is not correctly set. The expected value is 'RemoteOutbound' and the current value is"$TargetTenantOrgRelationship.MailboxMoveCapability -ForegroundColor Red
+        }
+    } else {
+        Write-Host ">> Error: Organization relationship on TARGET tenant mailbox is not enabled for moves" -ForegroundColor Red
+    }
+}
+
+function KillSessions {
+    #Check if there's any existing session opened for EXO and remove it so it doesn't remains open
+    Get-PSSession | Where-Object { $_.ComputerName -eq 'outlook.office365.com' } | Remove-PSSession
+}
+
+function CollectData {
+    #Create the folders based on date and time to store the files
+    $InputPath = Read-Host "Please specify an existing path to store the collected data "
+    $currentdate = (Get-Date).ToString('ddMMyyHHMM')
+    if (Test-Path $InputPath -PathType Container) {
+        $OutputPath = New-Item -ItemType Directory -Path $InputPath -Name $currentdate | Out-Null
+        $OutputPath = $InputPath + '\' + $currentdate
+    } else {
+        Write-Host ">> Error: The specified folder doesn't exist, please specify an existent one" -ForegroundColor Red
+        Exit
+    }
+
+    #Collect the Exchange Online data and export it to an XML file
+    Write-Host "Informational: Saving SOURCE tenant id to text file"  -ForegroundColor Yellow
+    "SourceTenantId: " + $SourceTenantId | Out-File $OutputPath\TenantIds.txt
+    Write-Host "Informational: Saving TARGET tenant id to text file"  -ForegroundColor Yellow
+    "TargetTenantId: " + $TargetTenantId | Out-File $OutputPath\TenantIds.txt -Append
+    Write-Host "Informational: Exporting the SOURCE tenant organization relationship"  -ForegroundColor Yellow
+    Get-SourceOrganizationRelationship | Export-Clixml $OutputPath\SourceOrgRelationship.xml
+    Write-Host "Informational: Exporting the TARGET tenant migration endpoint"  -ForegroundColor Yellow
+    Get-TargetMigrationEndpoint | Export-Clixml $OutputPath\TargetMigrationEndpoint.xml
+    Write-Host "Informational: Exporting the TARGET tenant organization relationship"  -ForegroundColor Yellow
+    Get-TargetOrganizationRelationship | Export-Clixml $OutputPath\TargetOrgRelationship.xml
+    Write-Host "Informational: Exporting the TARGET tenant accepted domains"  -ForegroundColor Yellow
+    Get-TargetAcceptedDomain | Export-Clixml $OutputPath\TargetAcceptedDomains.xml
+    Write-Host "Informational: Exporting the TARGET tenant Azure AD applications" - -ForegroundColor Yellow
+    Get-AzureADApplication | Export-Clixml $OutputPath\TargetAADApps.xml
+
+    #Compress folder contents into a zip file
+    Write-Host "Informational: Data has been exported. Compressing it into a ZIP file"  -ForegroundColor Yellow
+    if ((Get-ChildItem $OutputPath).count -gt 0) {
+        try {
+            Compress-Archive -Path $OutputPath\*.XML -DestinationPath $InputPath\CTMMCollectedData$currentdate.zip
```

Comment:
`Compress-Archive` is not always available in all PowerShell versions. Need to require a PowerShell version.